modules/gdscript/tests/scripts/analyzer/features/const_array_and_dictionary_constructors.gd

```diff
@@ -0,0 +1,35 @@
+const A = Array()
+const A1 = Array(Array())
+const A2 = Array([])
+const A3 = [Array()]
+const A4 = [[]]
+
+const IA = Array([1, 2, 3], TYPE_INT, &"", null)
+
+const D = Dictionary()
+const D1 = Dictionary(Dictionary())
+const D2 = Dictionary({})
+const D3 = {d = Dictionary()}
+const D4 = {d = {}}
```

Comment:
Would be good to add `var` counterparts ensuring they don't end up being readonly when initialized the same way (unless that's already tested elsewhere).

Reply:
Done.